includes/class-wc-moneybadger-gateway.php

```diff
@@ -0,0 +1,429 @@
+
+    /**
+	 * Generates the MoneyBadger iframe on the payment page.
+	 *
+	 * @param string $order_id  The order ID
+     * @return void
+	 */
+	public function receipt_page( $order_id ) {
+        $order = wc_get_order( $order_id );
+
+        wp_enqueue_script(
+			'moneybadger-payment-script',
+            plugins_url( './../assets/js/moneybadger-checkout.js', __FILE__ ),
+            array( 'jquery' )
+        );
+
+        $order_status_url_params = http_build_query([
+           "wc-api" => "wc_gateway_moneybadger_order_status",
+           "order_id" => $order_id,
+        ]);
+        $order_status_url = sprintf( "%s/?%s", home_url(), $order_status_url_params );
+
+        wp_localize_script(
+			'moneybadger-payment-script',
+			'moneybadger_checkout_params',
+			array(
+                'order_status_url' => $order_status_url,
+                'order_complete_url' => $this->get_return_url( $order ),
+			)
+		);
+
+        echo '<iframe src="' . esc_url (
+                $this->get_moneybadger_iframe_url( $order_id )
+            ) . '" width="100%" height="600" frameborder="0"></iframe>';
+    }
+
+	/**
+	 * Check if the gateway is available for use.
+	 *
+	 * @return bool
+	 */
+	public function is_available() {
+		if ( 'yes' === $this->enabled ) {
+            if ( get_woocommerce_currency() !== 'ZAR' ) {
+                return false;
+            }
+            if ( empty( $this->get_option('merchant_code') ) ) {
+                return false;
+            }
+            if ( empty( $this->get_option('api_key') ) ) {
+                return false;
+            }
+        }
+		return parent::is_available();
+	}
+
+    /**
+     * Check the MoneyBadger webhook
+     */
+    public function check_moneybadger_webhook() {
+        $payload = wp_unslash( $_GET );
+
+        if ( ! isset( $payload['reference'] ) ) {
+            wp_send_json_error( array( 'error' => 'Unexpected payload.' ) );
+        }
+
+        $reference = $payload['reference'];
+        // the reference is a concat of order id & timestamp
+        $order_id = explode( '-', $reference )[0];
+        $invoice = $this->get_invoice( $reference );
+
+        if ( is_wp_error( $invoice ) ) {
+            wp_send_json_error( array( 'error' => $invoice->get_error_message() ) );
+        }
+
+        $status = $invoice['status'];
+        $order = wc_get_order( $order_id );
+        if ( ! $order ) {
+            wp_send_json_error( array( 'error' => 'Order not found' ) );
+        }
+
+        if ( "AUTHORIZED" === $status ) {
+            $this->confirm_payment( $reference );
+            // fetch the status again after confirming
+            $invoice = $this->get_invoice( $reference );
+            if ( is_wp_error( $invoice ) ) {
+                wp_send_json_error( array( 'error' => $invoice->get_error_message() ) );
+            }
+            $status = $invoice['status'];
+        }
+
+        if ( "CONFIRMED" === $status ) {
+            if ( $order->is_paid() ) {
```

Comment:
please double check that the invoice amount matches the order amount, this prevents a whole host of potential vulns